pkg/utils/version.go

```diff
@@ -16,15 +16,16 @@ limitations under the License.
 
 package utils
 
-import "k8s.io/apimachinery/pkg/version"
+import (
+	"k8s.io/apimachinery/pkg/util/version"
+)
 
-const (
-	// KubeV1240 indicates the version of Kubernetes v1.24.0
-	// This should be "v1.24.0-alpha.4", but it does not tolerate v1.24.0.
-	KubeV1240 = "v1.24.0"
+var (
+	// KubeV1240Alpha4 indicates the version of Kubernetes v1.24.0-alpha.4
+	KubeV1240Alpha4 = version.MustParseSemantic("v1.24.0-alpha.4")
 )
 
 // SATokenAutoGenerated is used to indicate whether secret-based service account token can be auto generated
 func SATokenAutoGenerated(kubeServerVersion string) bool {
-	return version.CompareKubeAwareVersionStrings(kubeServerVersion, KubeV1240) > 0
+	return version.MustParseSemantic(kubeServerVersion).LessThan(KubeV1240Alpha4)
```

Comment:
`MustParseSemantic` will cause a panic error. Better wrap with a closure or handle the returned error.

Reply:
Updated.